packages/oui-stepper/src/step-form/step-form.controller.js

```diff
@@ -59,10 +60,11 @@ export default class StepFormController {
 
     onFormSubmit (form) {
         if (form.$valid && this.valid) {
-            this.onSubmit({ form });
-
-            // Focus next step
-            this.stepperCtrl.addForm(form, this.stepper.index);
+            const gotToNextStep = this.onSubmit({ form });
+            if (isNil(gotToNextStep) || gotToNextStep === true) {
+                // Focus next step
+                this.stepperCtrl.addForm(form, this.stepper.index);
+            }
```

Comment:
We could just do that I guess:
```javascript
if (form.$valid && this.valid && this.onSubmit({ form })) {
    this.stepperCtrl.addForm(form, this.stepper.index);
}
```

Reply:
I'm really not a fan of using a callback of a callback, to get a value we can get using `valid` attribute.
Can you provide more informations about  your use case (with code example) in the description of this PR ?

A thing we could still do here, is to call `this.onSubmit({ form });` outside the condition, to allow access of the `form` value even when the form is not valid.

Reply:
@euhmeuh, we can do that because in all the places where stepper is used, the onSubmit method does not return anything but still need to go to the next step. The condition to go to the next step is if nothing is returned or true is returned.

Reply:
@AxelPeter , i have added detail description with my usecase and code sample.

Reply:
@AxelPeter , I undo my changes and moved onSubmit call before if statement. This will be sufficient for my use case and looks very neat as well.